app/controllers/application_controller.rb

```diff
@@ -1,3 +1,14 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery
+
+  def require_user
+    redirect_to login_path unless current_user
+  end
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
+  end
+
+  helper_method :current_user
```

Comment:
I like to keep non-method-definitions above the method definitions, I find it is easier to know what is going on. You don't have to follow that convention of course :-)